src/panels/GrantSecretPanel/GrantSecretPanel.tsx

```diff
@@ -37,14 +38,10 @@ export type FormFields = {
   applications: string[];
 };
 
-const handleErrors = (response: string | ErrorResults) => {
-  if (typeof response === "string") {
-    throw new Error(response);
-  } else if (Array.isArray(response.results)) {
-    const error = response.results.find(({ error }) => !!error);
-    if (error?.error?.message) {
-      throw new Error(error.error?.message);
-    }
+const handleErrors = (response: ErrorResults) => {
+  const error = response.results.find(({ error }) => !!error);
+  if (error?.error?.message) {
+    throw new Error(error.error?.message);
   }
```

Comment:
Instead of me doing another PR, maybe you could fix this here with: 

```suggestion
  const errors = response.results.reduce((errorString, { error }) => {
    if (error?.message) {
      errorString = [errorString, error.message].filter(Boolean).join(". ");
    }
    return errorString;
  }, "");
  if (errors) {
    throw new Error(errors);
  }
```